Mathlib/GroupTheory/Nilpotent.lean

```diff
@@ -152,7 +152,13 @@ theorem upperCentralSeries_one : upperCentralSeries G 1 = center G := by
   ext
   simp only [upperCentralSeries, upperCentralSeriesAux, upperCentralSeriesStep, center, Set.center,
     mem_mk, mem_bot, Set.mem_setOf_eq]
-  exact forall_congr' fun y => by rw [mul_inv_eq_one, mul_inv_eq_iff_eq_mul, eq_comm]
+  constructor
+  · intro h
+    rw [← Set.mem_center_iff, Semigroup.mem_center_iff]
+    intro g
+    rw [eq_comm, ← mul_inv_eq_iff_eq_mul, ← mul_inv_eq_one, ← h g]
+  · intro h g
+    rw [mul_inv_eq_one, mul_inv_eq_iff_eq_mul, h.comm]
```

Comment:
Since we're working with an associative `Group G` anyway, we can keep the old proof. Or is this purposefully more complicated for a potential nonassociative generalization?
```lean
  ext
  simp only [upperCentralSeries, upperCentralSeriesAux, upperCentralSeriesStep,
    Subgroup.mem_center_iff, mem_mk, mem_bot, Set.mem_setOf_eq]
  exact forall_congr' fun y => by rw [mul_inv_eq_one, mul_inv_eq_iff_eq_mul, eq_comm]
```

Reply:
The original proof seems to work now so I have re-instated it. I think it didn't work for me in an earlier draft, but probably something else has been fixed in the meantime.

Reply:
I've re-instated the old proof (with `center, Set.center,` replaced with `Subgroup.mem_center_iff` in `simp only` as you suggest).